front/lib/api/assistant/conversation.ts

```diff
@@ -638,15 +649,117 @@ export async function* editUserMessage(
     message: UserMessageType;
     content: string;
   }
-): AsyncGenerator<UserMessageNewEvent | AgentErrorEvent> {
+): AsyncGenerator<UserMessageNewEvent | UserMessageErrorEvent> {
+  if (auth.user()?.id !== message.user?.id) {
+    return {
+      type: "user_message_error",
+      created: Date.now(),
+      messageId: message.sId,
+      error: {
+        code: "not_allowed",
+        message: "Only the author of the message can edit it",
+      },
+    };
+  }
+  const messageModel = await Message.findOne({
+    where: {
+      sId: message.sId,
+      conversationId: conversation.id,
+    },
+    include: [
+      {
+        model: UserMessage,
+        as: "userMessage",
+        required: true,
+      },
+    ],
+  });
+  if (!messageModel) {
+    return {
+      type: "user_message_error",
+      created: Date.now(),
+      messageId: message.sId,
+      error: {
+        code: "not_found",
+        message: "Message not found",
+      },
+    };
+  }
+  const userMessageModel = messageModel.userMessage;
+  if (!userMessageModel) {
+    return {
+      type: "user_message_error",
+      created: Date.now(),
+      messageId: message.sId,
+      error: {
+        code: "not_found",
+        message: "UserMessage not found",
+      },
+    };
+  }
+
+  const userMessage: UserMessageType = await front_sequelize.transaction(
+    async (t) => {
+      const m = await Message.create(
+        {
+          sId: messageModel.sId,
+          rank: messageModel.rank,
+          conversationId: conversation.id,
+          parentId: messageModel.parentId,
+          version: messageModel.version + 1,
```

Comment:
the retrieval of the messageRow shoud be in the same transaction otherwise you can have a race here